list_func.c

```diff
@@ -0,0 +1,192 @@
+#include <stdio.h>
+#include "list.h"
+
+/**
+ * @brief add a new element to the list at index
+ *
+ * @param root
+ * @param element
+ * @param index
+ */
+void add_element_at_index(t_list *root, t_list element, int index)
+{
+    t_list current;
+    int i;
+
+    i = 0;
+    if (*root)
+    {
+        current = *root;
+        while (current->next && i < index)
+        {
+            current = current->next;
+            i++;
+        }
+        if (i == index)
+        {
+            element->next = current;
+            element->previous = current->previous;
+            current->previous = element;
+            if (index != 0)
+                element->previous->next = element;
+            else
+                *root = element;
+        }
+        else
+            printf("Index Error\n");
+    }
+}
+
+/**
+ * @brief add a new element to the list before this element
+ *
+ * @param root
+ * @param new
+ * @param this
+ */
+void add_element_before_this(t_list *root, t_list new, t_list this)
+{
+    new->next = this;
+    if (this->previous)
+    {
+        new->previous = this->previous;
+        this->previous->next = new;
+        this->previous = new;
```

Comment:
C'est une idée d'utiliser this, mais du coup dans ce cas ça aurait dû être root qui s'appelle this non?

En OOP on utiliserait this pour celui à qui on fait l'opération, du coup ce serait plutôt:

my_list->add_element_before(before,new);

et du coup this, ce serait my_list.

C'est vrai que le nommage de la fonction détrompe, mais c'est un peu confus d'utiliser this pour autre chose que "this" en OOP

Reply:
oui, je comprends.